### PR TITLE
Add more linting/formatting rules

### PR DESCRIPTION
**Pull request by Nykseli**
## Detailed description
Having a good linting and formatting rules helps us keep the quality and style of the project good and consistent.
This PR tries to make those rules more clear.

Note that there was cases where the styles where slightly inconsistent so I did what ever felt right for me. I'm more than happy to modify/add/remove things, so please let me know your thoughts!

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/mangrove-lang/mangrove-vscode/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds and lints cleanly
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do
* [x] I've made sure that my commits are [signed/verified](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)

### CODE REVIEW

File: .eslintrc.json
Comment (freyjadomville) [inline .eslintrc.json:79]: We're probably gonna change this in a separate PR as the extensibility with commas is a benefit, I find. Unless you have justification for not changing this?

File: src/client/mangrove.ts
Comment (freyjadomville) [inline src/client/mangrove.ts:28]: This nullish coalescing to empty string feels unnecessary as the value will be falsy anyways

File: src/server/ast/symbolTable.ts
Comment (dragonmux) [inline src/server/ast/symbolTable.ts:164]: Don't love this.. we know `return` on its own returns undefined, and that was chosen to reduce the amount that could be typed wrong.. but, if you can provide us a compelling reason for this change, then sure

File: src/server/parser/error.ts
Comment (dragonmux) [inline src/server/parser/error.ts:93]: While strictly correct, we'd opted not to do this so the compiler would bark if the programmer forgot to add a new ErrorKind to this switch-case

File: src/server/parser/tokeniser.ts
Comment (dragonmux) [inline src/server/parser/tokeniser.ts:460]: This should probably generate a SyntaxError object so the system can then properly report this with an error squiggly down the road

**Review by freyjadomville — CHANGES_REQUESTED** [review]
We reviewed this in a pair with dragonmux, so her comments apply globally to ours

File: .eslintrc.json
Comment (Nykseli) [reply .eslintrc.json:79]: I chose the setting because it was more consistent with the current style. I'll keep this as is for now and create a new PR fixing it after this PR is merged. There's 1559 instances of `Missing trailing comma` so that way we can keep this PR smaller.

File: .eslintrc.json
Comment (Nykseli) [reply .eslintrc.json:17]: It can be safely dropped!

File: src/client/extension.ts
Comment (Nykseli) [reply src/client/extension.ts:49]: I'll revert the change. Reverting will also fix the issue with nullish coalescing @freyjadomville pointed out

File: .eslintrc.json
Comment (dragonmux) [reply .eslintrc.json:79]: We'll keep this as-is then for this PR and if you could please open a follow-up PR once this is merged to fix that, we'll then merge the results as we agree on PR size.

File: src/server/parser/error.ts
Comment (Nykseli) [reply src/server/parser/error.ts]: Now the `default` linting error is ignore for this specific case

**Comment by Nykseli** [comment]
All review items should now be addressed

File: src/server/parser/parser.ts
Comment (dragonmux) [inline src/server/parser/parser.ts:488]: Does this still have to be explicitly assigned undefined, or can the implicit undefined of `let` be used?

File: src/server/parser/tokeniser.ts
Comment (dragonmux) [reply src/server/parser/tokeniser.ts:461]: Ok, we can use the TODO to turn this into a SyntaxError too instead of console.warn()'ing, so we'll accept this.

**Review by dragonmux — COMMENTED** [review]
There is one question/query left to be answered before we approve for merge, but this is looking much better. Thank you!

File: src/server/parser/parser.ts
Comment (Nykseli) [reply src/server/parser/parser.ts]: It can be implicit. That was a mistake on my part. Good catch. Reverted the change

**Review by dragonmux — APPROVED** [review]
LGTM, once Freyja's also reviewed, we'll merge

**Review by freyjadomville — APPROVED** [review]
Also LGTM

**Pull request merged**
